### ISSUE TITLE
File Uploader 'Choose files to upload' button doesn't work in Firefox in some cirumstances

### ISSUE THREAD

**Issue by LeviWB** [silverstripe/silverstripe-framework#5417]
To replicate:
Firefox version tested: 45.0.2
1. Go to /admin and login
2. In the url bar, go directly to /admin/assets/add/?Locale=en_NZ&ID=0
3. Click the 'Choose files to upload...' button and select a file.

No file is uploaded and no feedback is provided. Uploading with the 'Drop Area' works fine.

When navigating to the file upload option using the CMS tabs, both upload buttons work fine.

Digging in a little deeper, the code from jquery.fileupload.js lines 729-730

```
            this.options.fileInput
                .bind('change.' + ns, {fileupload: this}, this._onChange);
```

The change event is not binded to the file upload input. It doesn't exist in the DOM when the bind is called for some reason.

A direct fix is to replace .bind with .on
`$(document).on('change.' + ns,  this.options.fileInput, {fileupload: this}, this._onChange);`

Obviously we don't want to modify third party plugins, but the jQuery File Upload plugin is pretty old, so upgrading may also solve the issue. I haven't looked into other workarounds outside the javascript.

**Comment by tractorcow** [comment]
Thanks @LeviWB , I've had a similar issue with this in other instances of uploadfield, not only in firefox.

The newer version of blueimp-fileupload has a fix for this so an upgrade might be enough.

https://github.com/blueimp/jQuery-File-Upload/blob/master/js/jquery.fileupload.js#L1287

**Comment by tractorcow** [comment]
So we are using "blueimp-file-upload": "^6.0.3", and need to upgrade to at LEAST 8.1. We may as well upgrade to 9.12 if we're doing an upgrade at all.

Upgrading in 3.x releases may be a bit problematic, since we haven't got a proper frontend dependency pipeline in place.

CC @clarkepaul @chillu as a possible alpha 2 task. Note that even though this issue references asset-admin, it's going to be an issue in other parts of the CMS where FileUpload is used. :D

**Comment by sb-relaxt-at** [comment]
I have encountered the same issue frequently, thanks for tracking it down! I am not sure if the suggested fix is really working (at least not in my setup). jQuery's "on" Function [expects a string as selector](http://api.jquery.com/on/) (and not a jQuery-/DOM-Node).

Given that any upload field has an associated id, one could write something like this:
`$(document).on('change.'+ns, '#'+this.options.fileInput.attr('id'), {fileupload: this}, this._onChange);`
In my limited testing this seems to solve the issue.

**Comment by tractorcow** [comment]
Upgrading to new fileupload is blocked by our jquery version. We'll need to upgrade to at least 1.10, it seems.

See https://github.com/silverstripe/silverstripe-framework/issues/3384 for ticket against upgrading jquery.

**Comment by tractorcow** [comment]
legacy uploadfield has been removed in 4.0. This will only be fixed in 3.x, if at all. Re-assigning.

**Comment by dhensby** [comment]
I'm afraid I don't see an upgrade of jQuery happening in 3.x

I'm going to close this as WONTFIX
